### PR TITLE
PWGHF: remove LF daughters propagation in Omegac0 reconstruction

### PR DESCRIPTION
**Pull request by ZFederica**

### CODE REVIEW

File: PWGHF/TableProducer/candidateCreatorOmegac.cxx [hunk 1]
```diff
@@ -184,81 +161,49 @@ struct HfCandidateCreatorOmegac {
...
-      dfv.propagateTracksToVertex();
-      if (!dfv.isPropagateTracksToVertexDone()) {
-        continue;
-      }
-      dfv.getTrack(0).getPxPyPzGlo(pvecV0Dau0);
-      dfv.getTrack(1).getPxPyPzGlo(pvecV0Dau1);
-      std::array<float, 3> pvecV0AsM = {pvecV0Dau0[0] + pvecV0Dau1[0], pvecV0Dau0[1] + pvecV0Dau1[1], pvecV0Dau0[2] + pvecV0Dau1[2]}; // AsM stands for "as mother"
+      // info from LF table
+      std::array<float, 3> pvecV0 = {v0Element.px(), v0Element.py(), v0Element.pz()}; // pvec stands for vector containing the 3-momentum components
+      std::array<float, 3> vertexV0 = {v0Element.x(), v0Element.y(), v0Element.z()};
```
Comment (vkucera) [inline PWGHF/TableProducer/candidateCreatorOmegac.cxx:175]:
```suggestion
      std::array<float, 3> pVecV0 = {v0Element.px(), v0Element.py(), v0Element.pz()}; // pvec stands for vector containing the 3-momentum components
      std::array<float, 3> vertexV0 = {v0Element.x(), v0Element.y(), v0Element.z()};
```

File: PWGHF/TableProducer/candidateCreatorOmegac.cxx [hunk 2]
```diff
@@ -184,81 +161,49 @@ struct HfCandidateCreatorOmegac {
...
+      const std::array<float, 6> covVtxV0 = {v0Element.positionCovMat()[0], v0Element.positionCovMat()[1], v0Element.positionCovMat()[2], v0Element.positionCovMat()[3], v0Element.positionCovMat()[4], v0Element.positionCovMat()[5]};
 
-      std::array<float, 3> coordVtxV0 = dfv.getPCACandidatePos();
-      std::array<float, 6> covVtxV0 = dfv.calcPCACovMatrixFlat();
+      std::array<float, 3> pvecV0Dau0 = {casc.pxpos(), casc.pypos(), casc.pzpos()};
+      std::array<float, 3> pvecV0Dau1 = {casc.pxneg(), casc.pyneg(), casc.pzneg()};
```
Comment (vkucera) [inline PWGHF/TableProducer/candidateCreatorOmegac.cxx:179]:
```suggestion
      std::array<float, 3> pVecV0Dau0 = {casc.pxpos(), casc.pypos(), casc.pzpos()};
      std::array<float, 3> pVecV0Dau1 = {casc.pxneg(), casc.pyneg(), casc.pzneg()};
```

**Review by vkucera — CHANGES_REQUESTED** [review]

File: PWGHF/TableProducer/candidateCreatorOmegac.cxx [hunk 3]
```diff
@@ -327,95 +272,78 @@ struct HfCandidateCreatorOmegac {
...
+        double pseudorapV0 = RecoDecay::eta(pvecV0);
 
-        // DCA between cascade daughters (from LF table)
-        double dcaCascDau = dfc.getChi2AtPCACandidate();
-        double dcaV0Dau = dfv.getChi2AtPCACandidate();
-        double dcaOmegacDau = df.getChi2AtPCACandidate();
+        // DCA between daughters
+        double dcaCascDau = casc.dcacascdaughters();
+        double dcaV0Dau = casc.dcaV0daughters();
+        double dcaOmegacDau = TMath::Sqrt(df.getChi2AtPCACandidate());
```
[the remark this answers is not in the document]
Comment (ZFederica) [reply PWGHF/TableProducer/candidateCreatorOmegac.cxx:307]: Done

File: PWGHF/DataModel/CandidateReconstructionTables.h [hunk 4]
```diff
@@ -1374,50 +1374,66 @@ DECLARE_SOA_COLUMN(ZDecayVtxCascade, zDecayVtxCascade, float);
 DECLARE_SOA_COLUMN(XDecayVtxV0, xDecayVtxV0, float);
 DECLARE_SOA_COLUMN(YDecayVtxV0, yDecayVtxV0, float);
 DECLARE_SOA_COLUMN(ZDecayVtxV0, zDecayVtxV0, float);
-DECLARE_SOA_COLUMN(SignDecay, signDecay, int);
+DECLARE_SOA_COLUMN(SignDecay, signDecay, int); // sign of pi <- xi
```
[the remark this answers is not in the document]
Comment (ZFederica) [reply PWGHF/DataModel/CandidateReconstructionTables.h:1377]: Done

File: PWGHF/TableProducer/candidateCreatorOmegac.cxx (hunk 2 above)
Comment (ZFederica) [reply PWGHF/TableProducer/candidateCreatorOmegac.cxx:179]: Done

File: PWGHF/TableProducer/candidateCreatorOmegac.cxx [hunk 5]
```diff
@@ -184,81 +161,49 @@ struct HfCandidateCreatorOmegac {
...
       // pseudorapidity
       double pseudorapPiFromCas = trackXiDauCharged.eta();
 
       // info from LF table
-      std::array<float, 3> vertexCascLFTable = {casc.x(), casc.y(), casc.z()};
+      std::array<float, 3> vertexCasc = {casc.x(), casc.y(), casc.z()};
+      std::array<float, 3> pvecCasc = {casc.px(), casc.py(), casc.pz()};
+      const std::array<float, 6> covVtxCasc = {casc.positionCovMat()[0], casc.positionCovMat()[1], casc.positionCovMat()[2], casc.positionCovMat()[3], casc.positionCovMat()[4], casc.positionCovMat()[5]};
+
+      std::array<float, 3> pvecPionFromCasc = {casc.pxbach(), casc.pybach(), casc.pzbach()};
```
[the remark this answers is not in the document]
Comment (ZFederica) [reply PWGHF/TableProducer/candidateCreatorOmegac.cxx:194]: Done

File: PWGHF/TableProducer/candidateCreatorOmegac.cxx (hunk 1 above)
Comment (ZFederica) [reply PWGHF/TableProducer/candidateCreatorOmegac.cxx:175]: Done

File: PWGHF/TableProducer/candidateCreatorOmegac.cxx (hunk 5 above)
[the remark this answers is not in the document]
Comment (ZFederica) [reply PWGHF/TableProducer/candidateCreatorOmegac.cxx:194]: Done

**Review by vkucera — APPROVED** [review]

**Pull request merged**